### PR TITLE
rework webhook receiver as a rules engine

### PR DESCRIPTION
**Pull request by eleith**
our config (not in this repo) contains all the specific rules we want for our own. this library is now left as a purely generic webhook reciever with more powerful options to edit and even slack on rule matches

### CODE REVIEW

File: lambda-functions/jira-webhook-receiver/config.js.txt [hunk 1]
```diff
@@ -8,13 +9,33 @@ module.exports = {
...
-    resolutionChannel: '#jira-resolution'
-    emptyComponentProjectExpression: /project5/i,
-    emptyComponentName: 'no-component',
-  }
+  rules: [
+    // add as many rules as needed
+    {
+      description: 'a rule description for logging',
+      // rules match on conditions in if
+      // conditions can be a string, regex or a function with an issue passed in
```
Comment (priyank-coursera) [inline lambda-functions/jira-webhook-receiver/config.js.txt]: a function that returns a boolean i guess

File: lambda-functions/jira-webhook-receiver/index.js [hunk 2]
```diff
@@ -1,227 +1,263 @@
...
+    values = array.map(element => element.value || element.name);
   }
 
-  const stakeHolders = getStakeHolders(issue);
-  const businessVerticals = getBusinessVerticals(issue);
+  return values;
+};
 
-  const groups = [];
+const testMatch = (valueTest, value, ...extras) => {
```
Comment (priyank-coursera) [inline lambda-functions/jira-webhook-receiver/index.js]: i would rename this from valueTest to rule/condition

Comment (priyank-coursera) [reply lambda-functions/jira-webhook-receiver/index.js]: checkRule(rule,value, ...extras) may be

File: lambda-functions/jira-webhook-receiver/config.js.txt (hunk 1 above)
Comment (eleith) [reply lambda-functions/jira-webhook-receiver/config.js.txt]: +1

File: lambda-functions/jira-webhook-receiver/index.js (hunk 2 above)
Comment (eleith) [reply lambda-functions/jira-webhook-receiver/index.js]: +1

**Pull request merged**